core/lib/Cardano/Address/Style/Shelley.hs

```diff
@@ -522,65 +522,75 @@ parseAddressInfoShelley AddressParts{..} = case addrType of
             { infoStakeReference = Just ByValue
             , infoSpendingKeyHash = Just addrHash1
             , infoStakeKeyHash = Just addrHash2
+            , infoAddressType = 0
```

Comment:
Rather than hard-coding numbers, can we just get `infoAddressType` using a bitwise "and" and right shift?

Reply:
good idea -> done